File: content/blog/counting_bits/go/pop_count.go
package counting_bits

func hammingWeight(n int) int {
	count := 0

	for n != 0 {
		count++
		n &= n - 1
	}

	return count
}

func countBitsPopCount(n int) []int {
	res := make([]int, 0, n+1)

	for i := 0; i <= n; i++ {
		res = append(res, hammingWeight(i))
	}

	return res
}

File: content/blog/counting_bits/ts/pop_count.ts
const hammingWeight = (n: number): number => {
    let count = 0

    while (n != 0) {
        count++
        n &= n - 1
    }

    return count
}

export const countBitsPopCount = (n: number): number[] => {
    const res: number[] = []

    for (let i = 0; i <= n; i++) {
        res.push(hammingWeight(i))
    }

    return res
}
